llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -4118,7 +4116,9 @@ SDValue AArch64TargetLowering::LowerFP_ROUND(SDValue Op,
         NaN = DAG.getNode(ISD::OR, dl, I32, Narrow,
                           DAG.getConstant(0x400000, dl, I32));
       }
-    } else if (SrcVT.getScalarType() == MVT::f64) {
+    } else if (SrcVT.getScalarType() == MVT::f64 && Subtarget->hasNEON()) {
+      assert((!VT.isVector() || Subtarget->isNeonAvailable()) &&
+             "FP_ROUND should have been scalarised");
```

Comment:
Is the scalar version of `FCVTXN` available in streaming mode? The pseudo code suggests not?